src/equals.ts

```diff
@@ -0,0 +1,41 @@
+/** @module ssz */
+import assert from "assert";
+import BN from "bn.js";
+
+import {
+  AnySSZType,
+  FullSSZType,
+  Type,
+} from "./types";
+import { assertValidValue } from "./assertValidValue";
+import { parseType } from "./util/types";
+
+
+/**
+ * Equality of two values of an SSZ type
+ */
+export function equals(value1: any, value2: any, type: AnySSZType): boolean {
```

Comment:
Wouldn't it make sense if `value1` and `value2` were `AnySSZType` rather than `any`?

Reply:
We've been letting input values be `any` (see `serialize` or `hashTreeRoot`).
At one point I tried to use a `SerializableValue` but that was causing issues.
The `AnySSZType` wouldn't really fit here, because its meant for types rather than values.

Reply:
that does make sense. Anyway you add the assertValidType in there. LGTM